### PR TITLE
Docs: flow control - Add a figure and fix suspected incorrect use of Envoy::Http::ConnectionManagerImpl

### PR DESCRIPTION
**Pull request by labilezhu**
Commit Message: docs: Add figure: Downstream connection backs-up and backpressure overview. fix incorrect use Envoy::Http::ConnectionManagerImpl
Additional Description:
Risk Level: na
Testing: na
Docs Changes: yes
Release Notes: na
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Review by adisuissa — COMMENTED** [review]
Great diagram, and thank you for contributing!
Assigning @KBaichoo who has domain expertise to validate correctness.
/assign @KBaichoo

File: source/docs/flow_control.md [hunk 1]
```diff
@@ -280,7 +283,7 @@ The high watermark path is as follows:
    `Network::ConnectionCallbacks::onAboveWriteBufferHighWatermark()`.
  * When `Envoy::Http::CodecClient` receives `onAboveWriteBufferHighWatermark()` it
    calls `onUnderlyingConnectionAboveWriteBufferHighWatermark()` on `codec_`.
- * When `Envoy::Http::ConnectionManagerImpl` receives `onAboveWriteBufferHighWatermark()` it calls
+ * When `Envoy::Http::Http2::ConnectionImpl` receives `onAboveWriteBufferHighWatermark()` it calls
```
Comment (KBaichoo) [inline source/docs/flow_control.md:286]: Thanks for catching this. Minor adjustment as both H1 and H2 connections will received `onUnderlyingConnectionAboveWriteBufferHighWaterMark` before further propagating.

File: source/docs/flow_control.md [hunk 2]
```diff
@@ -280,8 +283,8 @@ The high watermark path is as follows:
    `Network::ConnectionCallbacks::onAboveWriteBufferHighWatermark()`.
  * When `Envoy::Http::CodecClient` receives `onAboveWriteBufferHighWatermark()` it
    calls `onUnderlyingConnectionAboveWriteBufferHighWatermark()` on `codec_`.
- * When `Envoy::Http::ConnectionManagerImpl` receives `onAboveWriteBufferHighWatermark()` it calls
-   `runHighWatermarkCallbacks()` for each stream of the connection.
+ * When `Envoy::Http::Http2::ConnectionImpl` or `Envoy::Http::Http1::ConnectionImpl` receives `onAboveWriteBufferHighWatermark()` it calls
```
Comment (KBaichoo) [inline source/docs/flow_control.md:286]: one other detail is that  `Envoy::Http::Http2::ConnectionImpl` and `Envoy::Http::Http1::ConnectionImpl` receive

`onUnderlyingConnectionAboveWriteBufferHighWatermark` and `onUnderlyingConnectionBelowWriteBufferLowWatermark` not `onAboveWriteBufferHighWatermark`

which might have been part of the typo

**Review by KBaichoo — COMMENTED** [review]
Thank you. Just one minor change, otherwise lgtm

**Review by KBaichoo — APPROVED** [review]

**Pull request merged**